Refactor quick create antag panel code, fixed AFK revs not properly ending the round.

ShizCalev opened the pull request:
Fixes #29322

Also cleans up quick antag creation code a bit. 
Players who are AFK or zombies will no longer be made new antags through the panel.

ShizCalev committed:
Refactor quick create antag, fixed AFK revs not ending

Cyberboss commented on code/__HELPERS/game.dm:
```diff
@@ -330,6 +330,17 @@
 		afk_bracket += 600 // Add a minute to the bracket, for every attempt
 	return candidates
 
+/datum/proc/considered_alive(datum/mind/M)
```
certainly you meant something other than /datum?

Cyberboss commented on code/__HELPERS/game.dm:
```diff
@@ -330,6 +330,17 @@
+		return M.current.stat != DEAD && !issilicon(M.current) && !isbrain(M.current) && (!H || H.dna.species.id != "memezombies")
+	return FALSE
+
+/datum/proc/considered_afk(datum/mind/M)
```
ditto

ShizCalev committed:
More cleanup

ShizCalev committed:
Simplified one of the checks
The other half was covered by considered_alive already.

ShizCalev committed:
Reordered check

Eliminate current antags sooner before running them through the rest of
the checks.

AnturK commented on code/__HELPERS/game.dm:
```diff
@@ -330,6 +330,22 @@
 
+/proc/considered_alive(datum/mind/M, enforce_human = TRUE)
+	if(M && M.current)
+		switch(enforce_human)
```
All good but switch on boolean is just ugly.

ShizCalev committed:
removed switch
